src/Features/Core/IntroduceVariable/AbstractIntroduceVariableService.State.cs

```diff
@@ -216,9 +216,10 @@ private TExpressionSyntax GetExpressionUnderSpan(SyntaxTree tree, TextSpan textS
             private bool CanIntroduceVariable(
                 CancellationToken cancellationToken)
             {
-                // Don't generate a variable for an expression that's the only expression in a
-                // statement.  Otherwise we'll end up with something like "v;" which is not
-                // legal in C#.
+                // Don't generate a variable for an expression :
+                // (a) If that's the only expression in a statement.  
+                // Otherwise we'll end up with something like "v;" which is not legal in C#.
+                // (b) For Null Literals, as AllOccurences could introduce semantic errors.
```

Comment:
I would move the comment about why checking for null is important to the place where you are testing for null.